workers/unity/Packages/com.improbable.gdk.buildsystem/WorkerBuilder.cs

```diff
@@ -165,12 +196,18 @@ private static bool BuildWorkers(string[] workerTypes, BuildEnvironment buildEnv
             }
         }
 
-        private static bool BuildWorkerForEnvironment(string workerType, BuildEnvironment targetEnvironment, ScriptingImplementation? scriptingBackend = null)
+        private static bool BuildWorkerForEnvironmentTargets(
+            string workerType,
+            BuildEnvironment buildEnvironment,
+            IEnumerable<BuildTarget> buildTargets,
+            ScriptingImplementation? scriptingBackend = null)
         {
             var spatialOSBuildConfiguration = BuildConfig.GetInstance();
-            var environmentConfig = spatialOSBuildConfiguration.GetEnvironmentConfigForWorker(workerType, targetEnvironment);
+            var environmentConfig = spatialOSBuildConfiguration.GetEnvironmentConfigForWorker(workerType, buildEnvironment);
 
-            var enabledTargets = environmentConfig?.BuildTargets.Where(t => t.Enabled).ToList();
+            var enabledTargets = buildTargets.Any()
```

Comment:
This still feels wrong - an empty build targets array = build them all?

Reply:
I'd rather have a more comprehensive refactor to enable this behaviour

Reply:
Given the only real difference between passing in targets and inferring them is the contents of `enabledTargets`, I'm not really sure what the added value of splitting it out into different functions is